PWGGA/GammaConvBase/AliConversionPhotonCuts.cxx

```diff
@@ -4715,6 +4763,61 @@ UChar_t AliConversionPhotonCuts::DeterminePhotonQualityAOD(AliAODConversionPhoto
   return 0;
 }
 
+///________________________________________________________________________
+UChar_t AliConversionPhotonCuts::DeterminePhotonQualityTRD(AliAODConversionPhoton* photon, AliVEvent* eventDummy){
+
+  AliVTrack* negTrack = GetTrack(eventDummy, photon->GetTrackLabelNegative());
+  AliVTrack* posTrack = GetTrack(eventDummy, photon->GetTrackLabelPositive());
+
+  if(!negTrack || !posTrack) {
+      return 0;
+  }
+  if(negTrack->Charge() == posTrack->Charge()){
+      return 0;
+  }
+
+  Int_t nClusterTRDneg = negTrack->GetNcls(2);
+  Int_t nClusterTRDpos = posTrack->GetNcls(2);
+  
+  if (nClusterTRDneg > 1 && nClusterTRDpos > 1){
+    return 3;
+  } else if (nClusterTRDneg > 1 || nClusterTRDpos > 1){
+    return 2;
+  } else {
+    return 1;
+  }
+
+  return 0;
+}
+
+
+///________________________________________________________________________
+UChar_t AliConversionPhotonCuts::DeterminePhotonQualityTOF(AliAODConversionPhoton* photon, AliVEvent* eventDummy){
+
+  AliVTrack* negTrack = GetTrack(eventDummy, photon->GetTrackLabelNegative());
+  AliVTrack* posTrack = GetTrack(eventDummy, photon->GetTrackLabelPositive());
+
+  if(!negTrack || !posTrack) {
+      return 0;
+  }
+  if(negTrack->Charge() == posTrack->Charge()){
+      return 0;
+  }
+
+  Bool_t negTOFSignal = (negTrack->GetStatus() & AliVTrack::kTOFout ) && (negTrack->GetStatus() & AliVTrack::kTIME);
+  Bool_t posTOFSignal = (posTrack->GetStatus() & AliVTrack::kTOFout ) && (posTrack->GetStatus() & AliVTrack::kTIME);
+
+  if (negTOFSignal && posTOFSignal){
+    return 3;
+  } else if (negTOFSignal || posTOFSignal){
+    return 2;
+  } else {
+    return 1;
+  }
+
```

Comment:
return 0 will never be reached - is this intended?

Reply:
I had kept it because the return statements are also there in AliConversionPhotonCuts::DeterminePhotonQualiyAOD and AliConversionPhotonBase::DeterminePhotonQuality, but I have now removed the lines because they are indeed never reached.